Entities/User.php

```diff
@@ -1922,6 +1946,49 @@ public static function getUserSettings($return_hidden_column_name = false,
 
         return $list;
 
+    }
+    //-------------------------------------------------
+    public function verifySecurityAuthentication()
+    {
+
+        $this->security_code = null;
+        $this->security_code_expired_at = null;
+        $this->save();
+
+        $has_security = true;
+
+        $response['status'] = 'failed';
+        $response['data'] = null;
+
+        if(!config('settings.global.mfa_status')
+            || config('settings.global.mfa_status') === 'disable'
+            || !is_array(config('settings.global.mfa_methods'))
+            || (is_array(config('settings.global.mfa_methods'))
+            && count(config('settings.global.mfa_methods')) == 0)){
+            $has_security = false;
+        }
+
+        if(config('settings.global.mfa_status') == 'user-will-have-option'
+            && (!is_array($this->mfa_methods)
+                || (is_array($this->mfa_methods) && count($this->mfa_methods) == 0))){
+
+            $has_security = false;
+        }
+
+        if(!$has_security){
+            return $response;
+        }
+
+        $this->security_code = rand(100000, 999999);
+        $this->security_code_expired_at = now()->addMinutes(10);
+        $this->save();
+
+        dispatch(new SecurityOtpJob($this->toArray()));
```

Comment:
@we-vikram-wri231 this is also not correct, check other codes for reference

Reply:
@themodernpk Sir, I can't use VaahMail because we can't define the view in this, and I need to add a template for security OTP

Reply:
@we-vikram-wri231 discuss it with me